pkg/metrics/metrics.go

```diff
@@ -0,0 +1,30 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	/****Metrics related to Ark server backup***********/
+	lastBackupStatus = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "heptio-ark", // TODO: don't hard-code this value
```

Comment:
Given that the namespace needs to be flexible, I think it might make more sense to have a struct that holds on to all the metrics, e.g.
```go
type Metrics struct {
  lastBackupStatus prometheus.Guage
}

func NewMetrics(namespace string) *Metrics {
  return &Metrics{
    lastBackupStatus: prometheus.NewGuage(
      prometheus.GaugeOpts{
        Namespace: namespace,
        // ...
      },
    ),
  }
}

func (m *Metrics) UpdateLastBackupResult(success bool) {
  if success {
    m.lastBackupStatus.Set(1)
  } else {
    m.lastBackupStatus.Set(0)
  }
}
```

WDYT?

Reply:
Yes, that makes sense. I would have gone down that route to get rid of that `TODO` that I had.